api/files/views.py

```diff
@@ -162,3 +166,83 @@ def get_serializer_context(self):
         context = JSONAPIBaseView.get_serializer_context(self)
         context['file'] = self.file
         return context
+
+
+class FileMetadataRecordsList(JSONAPIBaseView, generics.ListAPIView, FileMixin):
+
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        PermissionWithGetter(ContributorOrPublic, 'target'),
+    )
+
+    required_read_scopes = [CoreScopes.NODE_FILE_READ]
+    required_write_scopes = [CoreScopes.NODE_FILE_WRITE]
+
+    serializer_class = FileMetadataRecordSerializer
+    view_category = 'files'
+    view_name = 'metadata-records'
+
+    def get_queryset(self):
+        return self.get_file().records.all()
+
+
+class FileMetadataRecordDetail(JSONAPIBaseView, generics.RetrieveUpdateAPIView, FileMixin):
+
+    record_lookup_url_kwarg = 'record_id'
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        PermissionWithGetter(ContributorOrPublic, 'target'),
+        PermissionWithGetter(ReadOnlyIfRegistration, 'target'),
+    )
+
+    required_read_scopes = [CoreScopes.NODE_FILE_READ]
+    required_write_scopes = [CoreScopes.NODE_FILE_WRITE]
+
+    serializer_class = FileMetadataRecordSerializer
+    view_category = 'files'
+    view_name = 'metadata-record-detail'
+
+    def get_object(self):
+        return utils.get_object_or_error(
+            self.get_file().records.filter(_id=self.kwargs[self.record_lookup_url_kwarg]),
+            request=self.request,
+        )
+
+
+class FileMetadataRecordDownload(JSONAPIBaseView, generics.RetrieveAPIView, FileMixin):
+
+    record_lookup_url_kwarg = 'record_id'
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        PermissionWithGetter(ContributorOrPublic, 'target'),
+    )
+
+    required_read_scopes = [CoreScopes.NODE_FILE_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    view_category = 'files'
+    view_name = 'metadata-record-download'
+
+    def get_serializer_class(self):
+        return None
+
+    def get_object(self):
+        return utils.get_object_or_error(
+            self.get_file().records.filter(_id=self.kwargs[self.record_lookup_url_kwarg]),
```

Comment:
I feel like we'll probably have a big "file metadata optimization party!" but just noticed this teeny one - could add `select_related('schema')` on this guy for the `get()` below